packages/babel-plugin/src/utils/evaluate-expression.tsx

```diff
@@ -205,6 +206,90 @@ const traverseFunction = (expression: t.Function, meta: Metadata) => {
   return createResultPair(value as t.Expression, updatedMeta);
 };
 
+/**
+ * Will find the function node for the call expression and wrap an IIFE around it (to avoid name collision)
+ * and move all the parameters mapped to passed arguments in the IIFE's scope (own scope in this case).
+ * It will also set own scope path so that when we recursively evaluate any node,
+ * we will look for its binding in own scope first, then parent scope.
+ *
+ * @param expression Expression we want to interrogate.
+ * @param meta Meta data used during the transformation.
+ */
+const traverseCallExpression = (expression: t.CallExpression, meta: Metadata) => {
+  const callee = expression.callee;
+  let value: t.Node | undefined | null = undefined;
+  let updatedMeta: Metadata = meta;
+
+  if (t.isExpression(callee)) {
+    let functionNode = null;
+
+    if (t.isIdentifier(callee)) {
+      const resolvedBinding = resolveBindingNode(callee.name, updatedMeta);
+
+      if (resolvedBinding && resolvedBinding.constant) {
+        functionNode = resolvedBinding.node;
+      }
+    } else if (t.isMemberExpression(callee) && t.isIdentifier(callee.property)) {
+      const { accessPath, bindingIdentifier } = getMemberExpressionMeta(callee);
+
+      if (bindingIdentifier) {
+        const resolvedBinding = resolveBindingNode(bindingIdentifier.name, updatedMeta);
+
+        if (resolvedBinding && resolvedBinding.constant) {
+          if (t.isObjectExpression(resolvedBinding.node)) {
+            functionNode = getValueFromObjectExpression(resolvedBinding.node, accessPath);
+          }
+        }
+      }
+    }
+
+    if (functionNode && t.isFunction(functionNode)) {
+      const { params } = functionNode;
+
+      const evaluatedArguments = expression.arguments.map(
+        (argument) => evaluateExpression(argument as t.Expression, updatedMeta).value
+      );
+
+      updatedMeta.parentPath.traverse({
+        CallExpression(callExpressionPath) {
+          // Only look for expression for which we are evaluating
+          if (callExpressionPath.node === expression) {
+            const callExpressionWrappingNode = wrapNodeInIIFE(callExpressionPath.node);
+
+            const [wrappingNodePath] = callExpressionPath.replaceWith(callExpressionWrappingNode);
+
+            wrappingNodePath.traverse({
+              ArrowFunctionExpression(arrowFunctionExpressionPath) {
+                params
+                  .filter((param) => t.isIdentifier(param) || t.isObjectPattern(param))
+                  .forEach((param, index) => {
+                    const evaluatedArgument = evaluatedArguments[index];
+
+                    arrowFunctionExpressionPath.scope.push({
```

Comment:
so we are doing this purely to add things to the scope? is this related to the ownPath hasOwnBinding logic?

Reply:
yups..I know its weird. So it creates an isolated scope and `ownPath` is only variable we use for getting the own binding because we are evaluating things recursively without any state in between.